legend-depot-store-mongo/src/main/java/org/finos/legend/depot/store/mongo/entities/EntitiesMongo.java

```diff
@@ -47,53 +44,38 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
-import java.util.Set;
+import java.util.Map;
 import java.util.StringTokenizer;
 import java.util.function.Consumer;
-import java.util.regex.Pattern;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 import static com.mongodb.client.model.Aggregates.group;
 import static com.mongodb.client.model.Filters.and;
 import static com.mongodb.client.model.Filters.eq;
 import static com.mongodb.client.model.Filters.not;
-import static com.mongodb.client.model.Filters.or;
 import static com.mongodb.client.model.Filters.regex;
 import static com.mongodb.client.model.Updates.combine;
 import static com.mongodb.client.model.Updates.currentDate;
 import static com.mongodb.client.model.Updates.set;
 import static org.finos.legend.depot.domain.version.VersionValidator.BRANCH_SNAPSHOT;
 
-public class EntitiesMongo extends BaseMongo<StoredEntity> implements Entities, UpdateEntities
+public class EntitiesMongo extends AbstractEntitiesMongo<StoredEntity> implements Entities<StoredEntity>, UpdateEntities<StoredEntity>
```

Comment:
the generic type should be T extends StoredEntity

Reply:
Entities mongo will only have StoredEntity and VersionedEntities mongo would be extending the AbstractMongo with StoredVersionedEntity

Reply:
made the change now